components/raftstore/src/store/worker/split_controller.rs

```diff
@@ -379,14 +383,39 @@ impl ReadStats {
         region_info.add_query_num(kind, query_num);
     }
 
-    pub fn add_flow(&mut self, region_id: u64, write: &FlowStatistics, data: &FlowStatistics) {
+    pub fn add_flow(
+        &mut self,
+        region_id: u64,
+        buckets: Option<&Arc<BucketMeta>>,
+        start: Option<&[u8]>,
+        end: Option<&[u8]>,
+        write: &FlowStatistics,
+        data: &FlowStatistics,
+    ) {
         let num = self.sample_num;
         let region_info = self
             .region_infos
             .entry(region_id)
             .or_insert_with(|| RegionInfo::new(num));
         region_info.flow.add(write);
         region_info.flow.add(data);
+        if let Some(buckets) = buckets {
+            let bucket_stat = self.region_buckets.entry(region_id).or_insert_with(|| {
+                let stats = new_bucket_stats(buckets);
+                BucketStat::new(buckets.clone(), stats)
+            });
+            let mut delta = metapb::BucketStats::default();
+            let bytes = write.read_bytes + data.read_bytes;
+            delta.set_read_bytes(vec![bytes as u64]);
+            let start = start.unwrap_or_default();
+            let end = end.unwrap_or_default();
+            merge_bucket_stats(
+                &bucket_stat.meta.keys,
```

Comment:
when is this meta.keys updated? It seems not updated once initialized. 
Also when parameter 'buckets' is changed (e.g. split buckets), will the bucket_stat.stats be re-created as well (otherwise the count won't be same)?

Reply:
Good catch :), buckets should be updated here.